_monitoring-your-cluster/logs.md

```diff
@@ -0,0 +1,205 @@
+
+OpenSearch has two *slow logs*, logs that help you identify performance issues: the search slow log and the indexing slow log.
+
+These logs rely on thresholds to define what qualifies as a "slow" search or indexing operation. For example, you might decide that a query is slow if it takes more than 15 seconds to complete. Unlike application logs, which you configure for modules, you configure slow logs for indexes. By default, both logs are disabled (all thresholds are set to `-1`):
+
+```json
+GET <some-index>/_settings?include_defaults=true
+
+{
+  "indexing": {
+    "slowlog": {
+      "reformat": "true",
+      "threshold": {
+        "index": {
+          "warn": "-1",
+          "trace": "-1",
+          "debug": "-1",
+          "info": "-1"
+        }
+      },
+      "source": "1000",
+      "level": "TRACE"
+    }
+  },
+  "search": {
+    "slowlog": {
+      "level": "TRACE",
+      "threshold": {
+        "fetch": {
+          "warn": "-1",
+          "trace": "-1",
+          "debug": "-1",
+          "info": "-1"
+        },
+        "query": {
+          "warn": "-1",
+          "trace": "-1",
+          "debug": "-1",
+          "info": "-1"
+        }
+      }
+    }
+  }
+}
+```
+
+To enable these logs, increase one or more thresholds:
+
+```json
+PUT <some-index>/_settings
+{
+  "indexing": {
+    "slowlog": {
+      "threshold": {
+        "index": {
+          "warn": "15s",
+          "trace": "750ms",
+          "debug": "3s",
+          "info": "10s"
+        }
+      },
+      "source": "500",
+      "level": "INFO"
+    }
+  }
+}
+```
+
+In this example, OpenSearch logs indexing operations that take 15 seconds or longer at the WARN level and operations that take between 10 and 14.*x* seconds at the INFO level. If you set a threshold to 0 seconds, OpenSearch logs all operations, which can be useful for testing whether slow logs are indeed enabled.
+
+- `reformat` specifies whether to log the document `_source` field as a single line (`true`) or let it span multiple lines (`false`).
+- `source` is the number of characters of the document `_source` field to log.
+- `level` is the minimum log level to include.
+
+A line from `opensearch_index_indexing_slowlog.log` might look like this:
+
+```
+node1 | [2019-10-24T19:48:51,012][WARN][i.i.s.index] [node1] [some-index/i86iF5kyTyy-PS8zrdDeAA] took[3.4ms], took_millis[3], type[_doc], id[1], routing[], source[{"title":"Your Name", "Director":"Makoto Shinkai"}]
+```
+
+Slow logs can consume considerable disk space if you set thresholds or levels too low. Consider enabling them temporarily for troubleshooting or performance tuning. To disable slow logs, return all thresholds to `-1`.
+
+## Task logs
+
+OpenSearch can log CPU time and memory utilization for the top N memory expensive search tasks when task resource consumers are enabled. By default, task resource consumers will log the top 10 search tasks at 60 second intervals. These values can be configured in `opensearch.yml`.
```

Comment:
I read this as a compound "N-memory-expensive search tasks". Maybe a second opinion for hyphenation?

Reply:
I just copied this file over to 1.3. I'll address the comments in another PR. Thanks!